webpack/assets/javascripts/react_app/components/ChartBox/ChartBox.js

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Card, Modal } from 'patternfly-react';
+import { isEqual } from 'lodash';
+import classNames from 'classnames';
+import { bindMethods } from '../../common/helpers';
+import DonutChart from '../common/charts/DonutChart';
+import BarChart from '../common/charts/BarChart';
+import { navigateToSearch } from '../../../services/charts/DonutChartService';
+import Loader from '../common/Loader';
+import MessageBox from '../common/MessageBox';
+import { translate as __ } from '../../common/I18n';
+
+class ChartBox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { showModal: false };
```

Comment:
can you add a comment here: `TODO: replace this with the modal infrastructure`

Reply:
I think we should have an issue in redmine for that and not add this as a comment to the code

Reply:
https://projects.theforeman.org/issues/28919